pkg/sql/colfetcher/colbatch_scan.go

```diff
@@ -140,9 +140,13 @@ func NewColBatchScan(
 
 	columnIdxMap := spec.Table.ColumnIdxMapWithMutations(returnMutations)
 	fetcher := cFetcher{}
+	if spec.IsCheck {
```

Comment:
I think the correct place to check this would be the `isSupported` check in `pkg/sql/colexec/colbuilder/execplan.go` which would allow us to only wrap the tablereader, rather than have the whole flow fall back to row execution cc @yuzefovich to confirm.